### PR TITLE
Fix the value issue of anthropic

### PR DESCRIPTION
**Pull request by shijiefengjun**
### What problem does this PR solve?

This pull request fixes the issue mentioned in https://github.com/infiniflow/ragflow/issues/3263.

1. response should be parsed as dict, prevent the following code from failing to take values:
ans = response["content"][0]["text"]
2. API Model ```claude-instant-1.2``` has retired (by [model-deprecations](https://docs.anthropic.com/en/docs/resources/model-deprecations)), it will trigger errors in the code, so I deleted it from the conf/llm_factories.json file and updated the latest API Model ```claude-3-5-sonnet-20241022```

### Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: rag/llm/chat_model.py [hunk 1]
```diff
@@ -1260,7 +1260,7 @@ def chat(self, system, history, gen_conf):
                 system=self.system,
                 stream=False,
                 **gen_conf,
-            ).json()
+            ).dict()
```
Comment (KevinHuSh) [inline rag/llm/chat_model.py]: Please refer to [this](https://www.geeksforgeeks.org/response-json-python-requests/)

**Review by KevinHuSh — CHANGES_REQUESTED** [review]
Please show us a reference.

**Comment by shijiefengjun** [comment]
> Please show us a reference.

You are rigorous. I read the following about response types in [anthropoic sdk python](https://github.com/anthropics/anthropic-sdk-python):
image

In fact, although ```.dict()``` is runnable, we should use ```.to_dict()``` instead. ```.to_dict()``` is already encapsulated in the anthropoic sdk, while ```.dict()``` calls pydantic lib methods.
anthropoic sdk:
image

pydantic lib: 
image

I have verified  ```.to_dict()``` in my local ragflow code, It's normal.

**Comment by KevinHuSh** [comment]
Appreciation!

**Pull request merged**